R/metab_simple.R

```diff
@@ -48,31 +54,39 @@ metab_simple <- function(data, ...) {
   #' 
   #' @param params a vector of length 2, where the first element is GPP and the second element is ER (both mg/L/d)
   #' @param 
-#   onestation_likelihood <- function(params, DO, DO.deficit, oxy, light, z, bp, ts, K) {
-#     
-#     GPP <- params[1]
-#     ER <- params[2]
-#     
-#     DO.modeled <- numeric(nrow(data))
-#     DO.modeled[1] <- oxy[1]
-#     for (i in 2:length(DO)) {
-#       DO.modeled[i] <- 
-#         DO.modeled[i-1] +
-#         (GPP/z)*(light[i]/sum(light)) + 
-#         ER * ts / z + 
-#         Kcor(temp[i],K)*ts*DO.deficit 
-#     }
-#   
-#     ##below is MLE calculation
-#     sqdiff<-(oxy-metab)^2 
-#     length(oxy)*(log(((sum(sqdiff)/length(oxy))^0.5)) +0.5*log(6.28))   + ((2*sum(sqdiff)/length(oxy))^-1)*sum(sqdiff)
-#   }
-  
+  onestation_negloglik <- function(params, DO.obs, DO.deficit, depth, k.O2, frac.GPP, frac.ER, frac.D) {
+    
+    # Parse params vector (passed from nlm)
+    GPP <- params[1]
+    ER <- params[2]
+    
+    # Model DO with given params
+    DO.mod <- numeric(length(DO.obs))
+    DO.mod[1] <- DO.obs[1]
+    DO.delta <- 
+      GPP * frac.GPP / depth + 
+      ER * frac.ER / depth + 
+      k.O2 * DO.deficit * frac.D # Bob - pros & cons of using DO.mod[t-1] instead of DO.obs[t] or DO.obs[t-1] (i.e., from DO.deficit) here?
```

Comment:
@jread-usgs @aappling-usgs @robohall 

I've found [this chapter by Bolker](http://ms.mcmaster.ca/~bolker/emdbook/chap11A.pdf) to be useful when thinking through some of the details in different model types.

But sometimes I still get confused, which is why I ask [questions like this](http://stats.stackexchange.com/questions/147991/assumptions-and-terminology-for-dynamic-regression-with-endogenous-offset-y-t). Note: I've clearly been thinking about this type of question a fair bit lately.